apps/bmd/src/AppRoot.tsx

```diff
@@ -139,880 +145,1063 @@ export const activationStorageKey = 'activation'
-      const { hardware } = this.props
-
-      // watch for changes
-      this.onDeviceChangeSubscription = hardware.devices
-        .pipe(map((devices) => Array.from(devices)))
-        .subscribe(async (devices) => {
-          const hasAccessibleControllerAttached = devices.some(
-            isAccessibleController
-          )
-          const hasCardReaderAttached = devices.some(isCardReader)
-          const printer = await hardware.readPrinterStatus()
-          const hasPrinterAttached = printer.connected
-
-          if (!hasPrinterAttached) {
-            this.resetBallot()
-
-            // stop+start forces a last-card-value cache flush
-            this.stopShortValueReadPolling()
-            this.startShortValueReadPolling()
-          }
-
-          this.setState({
+      },
+      GLOBALS.HARDWARE_POLLING_INTERVAL,
+      true
+    ),
+    [hardware]
+  )
+
+  // Handle Hardware Observer Subscription
+  useEffect(() => {
+    const hardwareStatusSubscription = hardware.devices
+      .pipe(map((devices) => Array.from(devices)))
+      .subscribe(async (devices) => {
+        const hasAccessibleControllerAttached = devices.some(
+          isAccessibleController
+        )
+        const hasCardReaderAttached = devices.some(isCardReader)
+        const newPrinter = await hardware.readPrinterStatus()
+        const hasPrinterAttached = newPrinter.connected
+        if (!hasPrinterAttached) {
+          resetBallot()
+          // stop+start forces a last-card-value cache flush
+          stopCardShortValueReadPolling()
+          startCardShortValueReadPolling()
+        }
+        dispatchState({
+          type: 'updateHardwareState',
+          hardwareState: {
             hasAccessibleControllerAttached,
             hasCardReaderAttached,
             hasPrinterAttached,
-          })
+          },
+        })
+      })
+    return () => hardwareStatusSubscription.unsubscribe()
+  }, [
+    hardware,
+    resetBallot,
+    startCardShortValueReadPolling,
+    stopCardShortValueReadPolling,
+  ])
+
+  // Handle Machine Config
+  useEffect(() => {
+    const setMachineConfig = async () => {
+      try {
+        const newMachineConfig = await machineConfigProvider.get()
+        dispatchState({
+          type: 'setMachineConfig',
+          machineConfig: newMachineConfig,
         })
+      } catch {
+        // Do nothing if machineConfig fails. Default values will be used.
+      }
     }
-  }
+    setMachineConfig()
+  }, [machineConfigProvider])
+
+  // Handle Keyboard Input
+  useEffect(() => {
+    document.documentElement.setAttribute('data-useragent', navigator.userAgent)
+    document.addEventListener('keydown', handleGamepadKeyboardEvent)
+    return () => {
+      document.removeEventListener('keydown', handleGamepadKeyboardEvent)
+    }
+  }, [])
+
+  // Bootstraps the AppRoot Component
+  useEffect(() => {
+    const retrieveVotes = () => storage.get(votesStorageKey)
```

Comment:
this should be gated in the same way on non-prod environment, to be safe... or again my preference would be to remove this.

Reply:
Why do we have this?

Reply:
This was to solve the issue of maintaining votes as the UI refreshes during development. 

Let's revisit after this PR.